spec/index.bs

```diff
@@ -642,36 +682,79 @@ interface GPUAdapter {
     **Arguments:**
         - optional {{GPUDeviceDescriptor}} |descriptor| = {}
 
-    **Returns:** |promise|, of type Promise<{{GPUDevice}}>
+    **Returns:** |promise|, of type Promise<{{GPUDevice}}>.
 
     Requests a [=device=] from the [=adapter=].
 
-    Returns [=a new promise=], |promise|.
-
-      - If the following requirements are met, |promise| [=resolves=] to a new
-        {{GPUDevice}} object encapsulating a new [=device=], which MUST
-        have exactly the capabilities described by |descriptor|.
+    Returns [=a new promise=], |promise|. Then, the following occurs asynchronously:
 
-        Issue: Figure out how to integrate the requirements.
+      - If the user agent can fulfill the request and
+        the [$requestDevice Valid Usage|valid usage$] rules are met,
+        |promise| [=resolves=] to a new {{GPUDevice}} object encapsulating
+        [=a new device=] with the capabilities described by |descriptor|.
 
       - Otherwise, |promise| [=rejects=] with an {{OperationError}}.
 </div>
 
+<div algorithm class=validusage>
+    <dfn abstract-op>requestDevice Valid Usage</dfn>
+
+    Given an [=adapter=] |adapter| and a {{GPUDeviceDescriptor}} |descriptor|,
+    the following validation rules apply:
+
+      - The set of {{GPUExtensionName}} values in
+        |descriptor|.{{GPUDeviceDescriptor/extensions}}
+        must be a subset of those in |adapter|.{{adapter/[[extensions]]}}.
+
+      - For each type of limit in {{GPULimits}}, the value of that limit in
+        |descriptor|.{{GPUDeviceDescriptor/limits}}
+        must be less than or equal to the value of that limit in
+        |adapter|.{{adapter/[[limits]]}}.
+</div>
+
+#### <dfn dictionary>GPUDeviceDescriptor</dfn> #### {#gpudevicedescriptor}
+
+{{GPUDeviceDescriptor}} describes a device request.
+
 <script type=idl>
 dictionary GPUDeviceDescriptor : GPUObjectDescriptorBase {
     sequence<GPUExtensionName> extensions = [];
     GPULimits limits = {};
-
-    // TODO: are other things configurable like queues?
 };
 </script>
 
+<dl dfn-type=dict-member dfn-for=GPUDeviceDescriptor>
+    : <dfn>extensions</dfn>
+    ::
+        The set of {{GPUExtensionName}} values in this sequence defines the exact set of
+        extensions that must be supported and allowed by a requested device.
+
+    : <dfn>limits</dfn>
+    ::
+        Defines the exact limits that must be supported and allowed by a requested device.
+</dl>
+
+#### <dfn enum>GPUExtensionName</dfn> #### {#gpuextensionname}
+
+Each {{GPUExtensionName}} identifies a set of functionality which, if available,
+allows additional usages of WebGPU that would have otherwise been invalid.
+
 <script type=idl>
 enum GPUExtensionName {
     "anisotropic-filtering"
 };
 </script>
 
+<dl dfn-type=enum-value dfn-for=GPUExtensionName>
+    : <dfn>"anisotropic-filtering"</dfn>
```

Comment:
Note for this PR, maybe we could replace it with "texture-compression-bc" that we know we want. Anisotropic filtering patents expired so we might be able to put it in the core WebGPU spec (and require all of our targets to support it, which is likely the case already)